### PR TITLE
Add script and Dockerfile to build custom Android package

### PR DESCRIPTION
**Pull request by edgchen1**
**Description**
Add script and Dockerfile to build custom Android package.

**Motivation and Context**
Provide an easier way to build a custom package.

### CODE REVIEW

**Comment by edgchen1** [comment]
Manually tested that the custom package built by this process works with an Android application.

File: tools/android_custom_build/readme.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+This directory contains helper files for building a custom ONNX Runtime
+Android package. It can be copied outside of the ONNX Runtime repo and used
+independently.
+
+Run the ./build_custom_android_package.py script. Use the --help option for
+more information.
+
```
Comment (skottmckay) [inline tools/android_custom_build/readme.txt:7]: Could we make this an md file?

Should we also include a step-by-step here or will that be added in the docs and we could link to it instead?

e.g. 
  - example of running model conversion script to get config to use
  - update build settings to enable type reduction/NNAPI if applicable
  - example command line to run with docker and copy out the aar
  - info or link to doco on how to use that in an android app

**Review by skottmckay — APPROVED** [review]
:shipit:

File: tools/android_custom_build/readme.txt (hunk 1 above)
Comment (edgchen1) [reply tools/android_custom_build/readme.txt]: added a link - there's a section there that we can add this information to:
https://onnxruntime.ai/docs/build/custom.html#android

**Review by skottmckay — APPROVED** [review]
:shipit:

**Pull request merged**